src/functions.c

```diff
@@ -549,10 +550,58 @@ static void fcallCommandGeneric(client *c, int ro) {
         return;
     }
 
+    if ((fi->f_flags & FUNCTION_FLAG_NO_CLUSTER) && server.cluster_enabled) {
+        addReplyError(c, "Can not run function on cluster, 'no-cluster' flag is set.");
+        return;
+    }
+
+    if ((fi->f_flags & FUNCTION_FLAG_DENY_OOM) && server.script_oom) {
+        addReplyError(c, "-OOM can not run the function");
```

Comment:
let's make the text part a bit more verbose and similar to the normal OOM error

Reply:
Fixed